cli/azd/internal/appdetect/java.go

```diff
@@ -43,165 +39,55 @@ func (jd *javaDetector) DetectProject(ctx context.Context, path string, entries
 		if strings.ToLower(entry.Name()) == "pom.xml" {
 			tracing.SetUsageAttributes(fields.AppInitJavaDetect.String("start"))
 			pomFile := filepath.Join(path, entry.Name())
-			project, err := readMavenProject(pomFile)
+			currentPom, err := toPom(pomFile)
```

Comment:
I'd recommend that we add one extra layer here to ensure the java project sees a maven project instead of a maven pom